### PR TITLE
[5.0] Pre2 - Convenience misc fixes.

### PR DESCRIPTION
**Pull request by frederikja163**
### Purpose of this PR
Fix some small errors in the generated spec.

* bools will now be booleans instead of bytes
* ref parameters can now be generic
* Delete\* will now recieve the same singular overloads as Gen\* and Create\*
* Added the ability to create differently named overloads
* Delete*, Gen* and Create* will have names with removed *s* for the singular versions

### Testing status
* Im looking at the generated code and compiling that for testing.

### CODE REVIEW

File: src/GeneratorV2/Process/Processor.cs [hunk 1]
```diff
@@ -475,7 +477,8 @@ public bool TryGenerateOverloads(Overload overload, [NotNullWhen(true)] out List
                     var returnType = new CSString(Nullable: true);
                     newOverloads = new List<Overload>()
                     {
-                        new Overload(overload, layer, overload.InputParameters, overload.NativeFunction, returnType, newReturnName, overload.GenericTypes)
+                        overload with{NestedOverload =  overload, MarshalLayerToNested = layer, ReturnType = returnType, ReturnVariableName = newReturnName},
```
Comment (NogginBops) [inline src/GeneratorV2/Process/Processor.cs:480]: Minor format detail but the Microsoft style puts a space between `with` and `{`.

File: src/GeneratorV2/Process/Processor.cs [hunk 2]
```diff
@@ -827,7 +828,8 @@ public bool TryGenerateOverloads(Overload overload, [NotNullWhen(true)] out List
                     var layer = new RefInsteadOfPointerLayer(changed, original);
                     newOverloads = new List<Overload>()
                     {
-                        new Overload(overload, layer, parameters, overload.NativeFunction, overload.ReturnType, overload.ReturnVariableName, overload.GenericTypes)
+                        overload with{NestedOverload = overload, MarshalLayerToNested = layer, InputParameters = parameters, GenericTypes = genericTypes},
```
Comment (NogginBops) [inline src/GeneratorV2/Process/Processor.cs:831]: Same thing here.

File: src/GeneratorV2/Process/Processor.cs [hunk 3]
```diff
@@ -871,22 +873,22 @@ public void WritePrologue(IndentedTextWriter writer)
...
+                if ((!nativeName.StartsWith("Create") && !nativeName.StartsWith("Gen") && !nativeName.StartsWith("Delete")) ||
                     !nativeName.EndsWith("s") || pointerParameter == null ||
                     pointerParameter.Type is not CSPointer pointerParameterType ||
                     pointerParameter.Length == null || pointerParameter.Length is not ParameterReference handleLength)
                 {
                     newOverloads = null;
                     return false;
                 }
-                var newNativeName = nativeName[..^1];
+                var newName = nativeName[..^1];
```
Comment (NogginBops) [inline src/GeneratorV2/Process/Processor.cs:891]: This should add a `NameMangler.RemoveEnd("s")` to make sure we are actually removing the right thing.

File: src/GeneratorV2/Process/Processor.cs [hunk 4]
```diff
@@ -335,7 +335,8 @@ public static BaseCSType MakeCSType(GLType type, string? group, out Expression?
                 case GLBaseType bt:
                     return bt.Type switch
                     {
-                        PrimitiveType.Void => new CSVoid(),
+                        PrimitiveType.Void => new CSVoid(bt.Constant),
+                        PrimitiveType.Bool => new CSType("bool", bt.Constant),
```
Comment (NogginBops) [inline src/GeneratorV2/Process/Processor.cs:339]: Are we sure this is getting marshalled correctly? Is `GLboolean` a byte or an int?

File: src/GeneratorV2/Process/Processor.cs (hunk 3 above)
Comment (frederikja163) [reply src/GeneratorV2/Process/Processor.cs:891]: We already check that the name ends with an "s" in the above if statement

File: src/GeneratorV2/Process/Processor.cs (hunk 4 above)
Comment (frederikja163) [reply src/GeneratorV2/Process/Processor.cs:339]: its a byte. Im not sure what you mean by marshalled correctly

Comment (NogginBops) [reply src/GeneratorV2/Process/Processor.cs:339]: In some flavors of C booleans are 4 bytes big. I'm just asking if the size is correct.

File: src/GeneratorV2/Process/Processor.cs (hunk 3 above)
Comment (NogginBops) [reply src/GeneratorV2/Process/Processor.cs:891]: Yeah ok, we might not catch extension methods using that though. But idk if there are any Gen, Create or Delete extension methods

**Review by NogginBops — APPROVED** [review]

**Comment by NogginBops** [comment]
These are some good additions!
Merging.

**Pull request merged**